src/full/Agda/TypeChecking/Rules/Display.hs

```diff
@@ -44,23 +44,28 @@ checkDisplayPragma f ps e = do
 --
 type M = ExceptT String TCM
 
-patternsToTerms :: Telescope -> [NamedArg A.Pattern] -> (Int -> Args -> M a) -> M a
+-- | Helper data type to record whether a pattern on the LHS contributed
+-- a 'Proj' elimination or an 'Apply' elimination.
+data ProjOrApp = IsProj QName | IsApp Term
+
+patternsToTerms :: Telescope -> [NamedArg A.Pattern] -> (Int -> Elims -> M a) -> M a
 patternsToTerms _ [] ret = ret 0 []
 patternsToTerms EmptyTel (p : ps) ret =
   patternToTerm (namedArg p) $ \n v ->
-  patternsToTerms EmptyTel ps     $ \m vs -> ret (n + m) (inheritHiding p v : vs)
+  patternsToTerms EmptyTel ps $ \m vs -> ret (n + m) (inheritHiding p v : vs)
```

Comment:
NB: The `$` is now also obsolete, thanks to `BlockArguments`.

Reply:
Should I clean this up while I'm here?

Reply:
I just meant since you already touched these lines, removing some whitespace...

I don't clean this up systematically, but when I change some code I also remove `$`s made redundant by `BlockArguments`.